integration-libs/epd-visualization/components/visual-viewer/visual-viewer.service.ts

```diff
@@ -0,0 +1,1496 @@
+  }
+  private set sceneId(value: string) {
+    this._sceneId = value;
+  }
+
+  private _contentType: ContentType;
+  private get contentType(): ContentType {
+    return this._contentType;
+  }
+  private set contentType(value: ContentType) {
+    this._contentType = value;
+  }
+
+  private _initialViewInfo: any;
+  private get initialViewInfo(): any {
+    return this._initialViewInfo;
+  }
+  private set initialViewInfo(value: any) {
+    this._initialViewInfo = value;
+  }
+
+  private _leafNodeRefs: NodeRef[];
+  private get leafNodeRefs(): NodeRef[] {
+    return this._leafNodeRefs;
+  }
+  private set leafNodeRefs(value: NodeRef[]) {
+    this._leafNodeRefs = value;
+  }
+
+  private _viewPriorToIsolateViewInfo: any;
+  private get viewPriorToIsolateViewInfo(): any {
+    return this._viewPriorToIsolateViewInfo;
+  }
+  private set viewPriorToIsolateViewInfo(value: any) {
+    this._viewPriorToIsolateViewInfo = value;
+  }
+
+  private _viewportAdded$: Observable<void>;
+  private get viewportAdded$(): Observable<void> {
+    return this._viewportAdded$;
+  }
+  private set viewportAdded$(value: Observable<void>) {
+    this._viewportAdded$ = value;
+  }
+
+  private _selectedNodeIds$ = new BehaviorSubject<string[]>([]);
+  private get selectedNodeIds$() {
+    return this._selectedNodeIds$;
+  }
+  private set selectedNodeIds$(value) {
+    this._selectedNodeIds$ = value;
+  }
+
+  private _sceneLoadInfo$ = new BehaviorSubject<SceneLoadInfo>({
+    sceneLoadState: SceneLoadState.NotStarted,
+  });
+  public get sceneLoadInfo$() {
+    return this._sceneLoadInfo$;
+  }
+
+  protected readonly DEFAULT_BACKGROUND_TOP_COLOR = '--cx-color-inverse';
+  protected readonly DEFAULT_BACKGROUND_BOTTOM_COLOR = '--cx-color-inverse';
+  protected readonly DEFAULT_HOTSPOT_SELECTION_HIGHLIGHT_COLOR =
+    'rgba(255, 0, 0, 0.6)';
+  protected readonly DEFAULT_SHOW_ALL_HOTSPOTS_COLOR = 'rgba(255, 255, 0, 0.3)';
+  protected readonly DEFAULT_OUTLINE_COLOR = 'red';
+  protected readonly DEFAULT_OUTLINE_WIDTH = 5;
+  protected readonly DEFAULT_SELECTION_MODE = SelectionMode.Exclusive;
+  protected readonly DEFAULT_SHOW_ALL_HOTSPOTS_ENABLED = false;
+  protected readonly DEFAULT_EXCLUDED_OPACITY = 0.2;
+  protected readonly DEFAULT_ZOOM_TO_MARGIN = 0.2;
+  protected readonly DEFAULT_FLY_TO_DURATION = 1;
+
+  private _flyToDurationInSeconds = this.DEFAULT_FLY_TO_DURATION;
+  private get flyToDurationInSeconds() {
+    return this._flyToDurationInSeconds;
+  }
+  private set flyToDurationInSeconds(value) {
+    this._flyToDurationInSeconds = value;
+  }
+  private _zoomToMargin = this.DEFAULT_ZOOM_TO_MARGIN;
+  private get zoomToMargin() {
+    return this._zoomToMargin;
+  }
+  private set zoomToMargin(value) {
+    this._zoomToMargin = value;
+  }
+
+  /**
+   * The top colour of the background gradient.
+   * Can be passed in the CSS color format or as a Spartacus theme color i.e. '--cx-color-background' with the quotes.
+   */
+  @Input()
```

Comment:
Could you help me understand why service contains `@Input`s, please?

Reply:
These annotations are not required. They are remnants from moving logic from the component to the service. I will remove them.